### ISSUE TITLE
AssestMarker gets removed when drawing the road

### ISSUE THREAD

**Issue by jesussmile** [liodali/osm_flutter#257]
Hi I was just trying out your `listenerMapSingleTapping.addListener` 
Works fine but my custom `AssetMarker` gets removed when the road is drawn, The only marker I get is from `roadOption , showMarkerofPoi:true` , plus the marker is only drawn over the roads which offsets the initial marker position. Its not a big issue and I am fine with what is right now but Is there a way I can still keep my initial markers ?
<img width="817" alt="road 2" src="https://user-images.githubusercontent.com/11044978/166866204-485dfa62-d69e-4fdb-a4af-8a8b8c583e65.png">

the code
```ruby
  osmController!.listenerMapSingleTapping.addListener(() async {
      if (osmController!.listenerMapSingleTapping.value != null) {
        if (pointManual1 == null) {
          pointManual1 = osmController!.listenerMapSingleTapping.value;
          List<Placemark> placemarks = await placemarkFromCoordinates(
              pointManual1!.latitude, pointManual1!.longitude);
          print(placemarks);
          Placemark placemark = placemarks[0];
          String? name = placemark.subLocality;
          print("sublocality " + name!);
          await osmController!.addMarker(pointManual1!,
              markerIcon: MarkerIcon(
                  assetMarker: AssetMarker(
                      image: AssetImage("images/images/pickicon.png"))));
        } else {
          pointManual2 = osmController!.listenerMapSingleTapping.value;
          List<Placemark> placemarks = await placemarkFromCoordinates(
              pointManual2!.latitude, pointManual2!.longitude);
          print(placemarks);
          await osmController!.addMarker(pointManual2!,
              markerIcon: MarkerIcon(
                  assetMarker: AssetMarker(
                      image: AssetImage("images/images/desticon.png"))));

          RoadInfo roadInformation = await osmController!.drawRoad(
              pointManual1!, pointManual2!,
              //interestPoints: [pointM1, pointM2],
              roadOption: RoadOption(
                  roadWidth: 10,
                  roadColor: Colors.blue,
                  zoomInto: true,
                  showMarkerOfPOI: true));

          String duration =
              "duration: ${Duration(seconds: roadInformation.duration!.toInt()).inMinutes} minutes ";
          String distance =
              "distance: ${roadInformation.distance!.toStringAsFixed(2)} Km";
          print(duration);
          print(distance);
          ScaffoldMessenger.of(context).showSnackBar(SnackBar(
            duration: Duration(seconds: 10),
            content: Text(duration + distance),
          ));
        }
      }
    });
```

**Comment by liodali** [comment]
i will remove that behaviour from drawRoad method

**Comment by Eimji** [comment]
Hello,

I also notice that the user location marker also disappears when the road is drawn (I set `trackMyPosition: true,` in OSMFlutter).
Waiting for your fix in 0.35.0 :)
Thanks
